app/domain/authentication/authn_gce/extract_resource_restrictions.rb

```diff
@@ -0,0 +1,86 @@
+require 'command_class'
+
+module Authentication
+  module AuthnGce
+
+    # This class is responsible of restrictions extraction that are set on a Conjur host or user as annotations.
+    ExtractResourceRestrictions = CommandClass.new(
+      dependencies: {
+        role_class:              ::Role,
+        resource_class:          ::Resource,
+        validate_account_exists: ::Authentication::Security::ValidateAccountExists.new,
+        logger:                  Rails.logger
+      },
+      inputs:       %i(account username extraction_prefix)
+    ) do
+
+      def call
+        validate_account_exists
+        extract_resource_restrictions
+        resource_restrictions
+      end
+
+      private
+
+      def validate_account_exists
+        @validate_account_exists.(
+          account: @account
+        )
+      end
+
+      def extract_resource_restrictions
+        @logger.debug(
+          LogMessages::Authentication::AuthnGce::ExtractingRestrictionsFromResource.new(
+            @extraction_prefix,
+            @username
+          )
+        )
+        prefixed_resource_annotations.select do |a|
+          annotation_name = a.values[:name]
+          resource_value = annotation_value(annotation_name)
+          next unless resource_value
+          resource_restrictions.push(
+            ResourceRestriction.new(
```

Comment:
Reason why this is called ResourceRestriction and not `GCEResource`?

Reply:
if we want to make it common, then let's take it out of the module AuthnGce, otherwise IMO it is cleaner to label it as gce

Reply:
it may go outside the module in the future but we didn't want to do it now. i think that `GCEResource` is redundant as it's under the `AuthnGce` module (so it's `AuthnGCE::GCEResource`).

However, i do prefer `AuthnGce::Resource` rather than `AuthnGce::ResourceRestriction` as this describes the resource itself (the restriction is on the host).

@nessiLahav what do you think?

Reply:
I like `ResourceRestriction` name more, because i think it is make the code more understandable as this resource is an actually a restriction.

But we can challenge it again at the moment we will refactor it to be outside of the module

Reply:
it is not really a restriction but it's not important enough for me to challenge this.

Reply:
`ResourceProperty`?

it will be used for `project-id`, `instance-name`, `service-account-id` and `service-account-email` which are all properties of a GCE resource.

But i can really live with `ResourceRestriction`.

Reply:
no problem 💯